src/Data/Star/Environment.agda

```diff
@@ -43,5 +43,4 @@ Env T Γ = All T Γ
 -- A safe lookup function for environments.
 
 lookup : ∀ {Γ σ} {T : Ty → Set} → Env T Γ → Γ ∋ σ → T σ
-lookup ρ i with Pointer.lookup ρ i
-... | result refl x = x
+lookup ρ i = case Pointer.lookup ρ i of λ where (result refl x) → x
```

Comment:
Compare ...
```agda
lookup ρ i with result refl x ← Pointer.lookup ρ i = x
```
... I'd go with irrefutable `with` here!

Reply:
Yes, it does indeed improve things! Done.